rust/worker/src/segment/types.rs

```diff
@@ -112,20 +112,25 @@ impl ChromaError for LogMaterializerError {
     }
 }
 
-#[derive(Debug, Clone)]
-pub struct MaterializedLogRecord<'referred_data> {
+/// This struct is used internally. It is not exposed to materialized log consumers.
+///
+/// Instead of cloning or holding references to log records/segment data, this struct contains owned values that can be resolved to the referenced data.
+/// E.x. `data_record_offset_id: Option<u32>` is used instead of `data_record: Option<&DataRecord>` to avoid holding references to the data.
+/// This allows `MaterializedLogRecord` (and types above it) to be trivially Send'able.
+#[derive(Debug)]
+struct MaterializedLogRecord {
     // This is the data record read from the record segment for this id.
     // None if the record exists only in the log.
-    pub(crate) data_record: Option<DataRecord<'referred_data>>,
+    data_record_offset_id: Option<u32>,
```

Comment:
can `data_record_offset_id` and `offset_id` can be combined into one? It seems like one of them is a subset of the other

Reply:
good point, we need this field to know whether the offset ID exists in the segment but it can be a simple bool instead of storing the same offset for clarity

e4f641992